frontend/packages/console-plugin-sdk/src/typings/features.ts

```diff
@@ -4,42 +4,34 @@ import { Extension, AlwaysOnExtension } from './base';
 import { Action } from 'typesafe-actions';
 
 namespace ExtensionProperties {
-  interface FeatureFlag {
-    /** The name of the feature flag. */
+  export interface ModelFeatureFlag {
     flag: string;
-  }
-
-  export interface ModelFeatureFlag extends FeatureFlag {
     /** If a CRD for this model exists, the feature will be enabled. */
     model: K8sKind;
   }
 
-  export interface ActionFeatureFlag extends FeatureFlag {
-    /** Function used to detect the feature and set flag name/value via Redux action dispatch. */
-    detect: ActionFeatureFlagDetector;
+  export interface FeatureDetect {
+    /** Function used to detect the feature and set arbitrary flag name/value via Redux action dispatch. */
+    detect: ActionFeatureDetector;
   }
 }
 
 export interface ModelFeatureFlag extends AlwaysOnExtension<ExtensionProperties.ModelFeatureFlag> {
   type: 'FeatureFlag/Model';
 }
 
-export interface ActionFeatureFlag extends Extension<ExtensionProperties.ActionFeatureFlag> {
-  type: 'FeatureFlag/Action';
+export interface FeatureDetect extends Extension<ExtensionProperties.FeatureDetect> {
+  type: 'FeatureDetector/Action';
 }
 
-export type FeatureFlag = ModelFeatureFlag | ActionFeatureFlag;
+export type FeatureFlag = ModelFeatureFlag;
```

Comment:
This was meant to union-ize extension types which shared some common aspects.

Since it's no longer the case here, I'd suggest to remove the `FeatureFlag` type alias.